pyccel/codegen/printing/ccode.py

```diff
@@ -2298,7 +2298,8 @@ def _print_FunctionCall(self, expr):
 
         args += self._temporary_args
         self._temporary_args = []
-        args = ', '.join(['{}'.format(self._print(a)) for a in args])
+        args = ', '.join(self._print(ai) for a in args for ai in \
+                        ([a] if isinstance(a, (FunctionAddress, Nil)) else flatten_tuple_var(a, self.scope)))
```

Comment:
In the case of a `FunctionAddress`, doesn't `flatten_tuple_var(a)` return `[a]` as wanted?

I see that it works differently in the case of `Nil`, though...

Reply:
No but that would be a better way to handle this :+1:

Reply:
Fixed : ee9cb2f and [69c5afa](https://github.com/pyccel/pyccel/pull/2232/commits/69c5afa3f1053af8f26bc42e74b64e14bc3470ee)